system-addon/test/unit/lib/TelemetryFeed.test.js

```diff
@@ -0,0 +1,262 @@
+    globals.set("ClientID", {getClientID: sandbox.spy(async () => FAKE_TELEMETRY_ID)});
+    globals.set("gUUIDGenerator", {generateUUID: () => FAKE_UUID});
+    instance = new TelemetryFeed();
+    instance.store = store;
+  });
+  afterEach(() => {
+    globals.restore();
+  });
+  describe("#init", () => {
+    it("should add .telemetrySender, a TelemetrySender instance", async () => {
+      assert.isNull(instance.telemetrySender);
+      await instance.init();
+      assert.instanceOf(instance.telemetrySender, TelemetrySender);
+    });
+    it("should add .telemetryClientId from the ClientID module", async () => {
+      assert.isNull(instance.telemetryClientId);
+      await instance.init();
+      assert.equal(instance.telemetryClientId, FAKE_TELEMETRY_ID);
+    });
+  });
+  describe("#addSession", () => {
+    it("should add a session", () => {
+      addSession("foo");
+      assert.isTrue(instance.sessions.has("foo"));
+    });
+    it("should set the start_time", () => {
+      sandbox.spy(Components.utils, "now");
+      const session = addSession("foo");
+      assert.calledOnce(Components.utils.now);
+      assert.equal(session.start_time, Components.utils.now.firstCall.returnValue);
+    });
+    it("should se the session_id", () => {
+      sandbox.spy(global.gUUIDGenerator, "generateUUID");
+      const session = addSession("foo");
+      assert.calledOnce(global.gUUIDGenerator.generateUUID);
+      assert.equal(session.session_id, global.gUUIDGenerator.generateUUID.firstCall.returnValue);
+    });
+    it("should set the page", () => {
+      const session = addSession("foo");
+      assert.equal(session.page, "NEW_TAB"); // This is hardcoded for now.
+    });
+  });
+  describe("#endSession", () => {
+    it("should not throw if there is no session for the given port ID", () => {
+      assert.doesNotThrow(() => instance.endSession("doesn't exist"));
+    });
+    it("should add a session_duration", () => {
+      sandbox.stub(instance, "sendEvent");
+      const session = addSession("foo");
+      instance.endSession("foo");
+      assert.property(session, "session_duration");
+    });
+    it("should remove the session from .sessions", () => {
+      sandbox.stub(instance, "sendEvent");
+      addSession("foo");
+      instance.endSession("foo");
+      assert.isFalse(instance.sessions.has("foo"));
+    });
+    it("should call createSessionSendEvent and sendEvent with the sesssion", () => {
+      sandbox.stub(instance, "sendEvent");
+      sandbox.stub(instance, "createSessionEndEvent");
+      const session = addSession("foo");
+      instance.endSession("foo");
+
+      // Did we call sendEvent with the result of createSessionEndEvent?
+      assert.calledWith(instance.createSessionEndEvent, session);
+      assert.calledWith(instance.sendEvent, instance.createSessionEndEvent.firstCall.returnValue);
+    });
+  });
+  describe("ping creators", () => {
+    beforeEach(async () => await instance.init());
```

Comment:
I'm confused.  This looks like it declares an anonymous function which can't ever be called, since there's no name or reference to call it by.  Am I missing something?